<a href="https://colab.research.google.com/github/alvinhugo/studi-kasus-02-kelompok-03/blob/main/Studi_Kasus_2_Kelompok_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**STUDI KASUS 2 KELOMPOK 3**
1. Puput Hanifah Nuranisa     (11210940000017)
2. Alvinhugo Zuhdi Nurharyadi (11210940000031)
3. Rifaldi Achmad Faisal (11210940000027)
4. Nurliana (11210940000026)
5. Becky Karlina Sindy (11210940000013)
6. M. Ikhwan Farhat (11210940000007)

In [ ]:
import pandas as pd
import seaborn as sns

# Loading Data:
try: # Running Locally, yakinkan "file_" berada di folder "data"
    mengajar = pd.read_csv("/content/mengajar.csv", low_memory = False, encoding='utf8')
    dosen = pd.read_csv("/content/tb_dosen.csv", low_memory = False, encoding='utf8')
    kelas = pd.read_csv("/content/tb_kelas.csv", low_memory = False, encoding='utf8')
    krs = pd.read_csv("/content/tb_krs.csv", low_memory = False, encoding='utf8')
    mhs = pd.read_csv("/content/tb_mahasiswa.csv", low_memory = False, encoding='utf8')
    matkul = pd.read_csv("/content/tb_mata_kuliah.csv", low_memory = False, encoding='utf8')
    nilai = pd.read_csv("/content/tb_nilai.csv", low_memory = False, encoding='utf8')
except: # Running in Google Colab
    !mkdir data
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/mengajar.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_dosen.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_kelas.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_krs.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_mahasiswa.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_mata_kuliah.csv
    !wget -P data/ https://github.com/alvinhugo/studi-kasus-02-kelompok-03/blob/main/tb_nilai.csv

In [ ]:
#1
nilai_dosen = nilai.groupby(nilai['NIDN']).mean()[['nilai']]
nilai_dosen = nilai_dosen.rename({'nilai': 'mean_nilai'}, axis=1) 
nilai_dosen = pd.merge(dosen, nilai_dosen, left_on='NIDN', right_index=True)
nilai_dosen = nilai_dosen[['NIDN', 'nama', 'mean_nilai']]
sb = nilai.groupby(nilai['NIDN']).std()[['nilai']]
sb = sb.rename({'nilai': 'std_nilai'}, axis=1) 
nilai_dosen = nilai_dosen.set_index(['NIDN']) 
nilai_dosen = pd.concat([nilai_dosen, sb], axis=1)
jml = pd.crosstab(nilai.NIDN, nilai.id_mata_kuliah)
total = pd.merge(nilai_dosen, jml, left_on='NIDN', right_index=True)
total

nama  mean_nilai  std_nilai  CS01  KAL01  KAL02  STA02
NIDN                                                                 
123    Andri Basuki   85.750000   7.741355     0      8      0      0
234        zulfikar   80.428571  15.883504     7      0      0      0
345   Annisa Fitria   60.090909  16.220077     0      0      4      7
787            yudi   70.571429  13.010984     0      7      0      0

# **Interpretasi**

Berdasarkan hasil olah data, Andri Basuki diketahui cenderung memberi nilai lebih tinggi dibandingkan dengan dosen lain dengan ukuran penyebaran datanya relatif kecil dan Andri Basuki mengajar di kelas Kalkulus Dasar dengan 8 mahasiswa.
Sedangkan Annisa Fitria cenderung memberi nilai rendah di bandingkan dosen lain dengan ukuran penyebaran datanya relatif kecil juga. Annisa Fitria mengajar di mata kuliah Kalkulus Lanjut dengan 4 mahasiswa dan Statistika Elementer dengan 7 mahasiswa. Jika dilihat dari jumlah mahasiswa yang diajarkan, Annisa fitria mengajar lebih banyak mahasiswa dibandingkan Andri Basuki.

In [ ]:
#2
dosen = dosen[['NIDN','Pendidikan_Terakhir']]
dosen = pd.merge(dosen, total, left_on='NIDN', right_index = True)
dosen = dosen[['Pendidikan_Terakhir', 'mean_nilai']]
group = dosen['mean_nilai'].groupby(dosen['Pendidikan_Terakhir'])
group.mean()

Pendidikan_Terakhir
S2    65.331169
S3    83.089286
Name: mean_nilai, dtype: float64

# **Interpretasi**

Berdasarkan hasil olah data, dosen dengan pendidikan terakhir S3 cenderung memberikan nilai yang lebih tinggi dibanding dosen dengan pendidikan terakhir S2. Hal ini ditunjukkan dari rata-rata dosen berpendidikan terakhir S3 adalah sebesar 83.089286, sedangkan dosen berpendidikan terakhir S2 menunjukkan rata-rata nilai sebesar 65.331169

In [ ]:
#3
nilai1 = nilai.groupby(nilai['nim']).mean()[['nilai']]
nilai1 = pd.merge(mhs, nilai1, left_on='nim',right_index=True)
grouped = nilai1['nilai'].groupby(nilai1['Asal_Sekolah'])
grouped.mean()

Asal_Sekolah
MA     71.937500
SMK    85.900000
SMU    66.416667
Name: nilai, dtype: float64

# **Interpretasi**

Berdasarkan hasil olah data, terdapat perbedaan performa akademik antara mahasiswa dari Madrasah/SMK dan SMU. Hal ini ditunjukkan melalui rata-rata nilai mahasiswa yang berasal dari MA, SMK, dan SMU. Ditunjukkan bahwa mahasiswa dari SMU mempunyai rata-rata nilai terkecil, yaitu 66.416667. Mahasiswa dari MA mempunyai rata-rata nilai 71.937500. Terakhir, mahasiswa dari SMK mempunyai rata-rata nilai terbesar, yaitu 85.9

In [ ]:
#4
G = pd.merge(mhs, nilai, on="nim")
G = G[['nim', 'nama', 'Gender', 'nilai']]
grouped = G['nilai'].groupby(G['Gender'])
grouped.mean()

Gender
Pria      73.888889
Wanita    71.600000
Name: nilai, dtype: float64

# **Interpretasi**

Berdasarkan hasil olah data, terdapat perbedaan pada kemampuan akademik antara mahasiswa dan mahasiswi. Hal ini ditunjukkan oleh rata-rata nilai. Untuk mahasiswa mempunyai rata-rata nilai sebesar 73.888889, sedangkan untuk mahasiswi mempunyai rata-rata nilai sebesar 71.6

In [ ]:
#5

#Variabel nilai
nilai3 = nilai.groupby(nilai['nim']).mean()[['nilai']]

#Variabel Angkatan
nilai4 = mhs.groupby(mhs['nim']).mean()[['Angkatan']]

#Penggabungan Dua DataFrame
nilai_akt = pd.merge(nilai3, nilai4, left_on='nim', right_index=True)
grouped = nilai_akt['nilai'].groupby(nilai_akt['Angkatan'])
grouped.mean()

Angkatan
2019.0    75.59375
2020.0    73.00000
Name: nilai, dtype: float64

# **Interpretasi**

Berdasarkan hasil olah data, ditunjukkan bahwa terdapat perbedaan prestasi akademik data antar angkatan mahasiswa. Hal ini terlihat bahwa untuk mahasiswa angkatan 2020 mempunya rata-rata nilai sebesar 73, sedangkan mahasiswa angkatan 2019 mempunyai rata-rata nilai sebesar 75.59375

# **Permasalahan**

- Terdapat beberapa data yang tidak diperlukan untuk mengolah data tersebut
- Terdapat banyak variabel yang tidak dibutuhkan
- Terdapat variabel yang memuat isi yang sama, namun penamaan variabelnya berbeda

# **KESIMPULAN**

Kesimpulan yang dapat kita ambil dari pengolahan data yang sudah dilakukan yaitu


*   Ada kecenderungan dosen memberikan nilai yang lebih tinggi maupun rendah.
*   Pendidikan terakhir dari seorang dosen mempengaruhi penilaiannya untuk mahasiswa, tetapi hal ini masih kurang akurat dan kredibel karena data yg kami olah sedikit.
*   Perbedaan asal sekolah mahasiswa mempengaruhi performa akademiknya, yaitu dilihat dari perolehan rata-rata nilainya.
*   Perbedaan jenis kelamin mahasiswa sedikit mempengaruhi performa akademiknya, yaitu dilihat dari perolehan rata-rata nilainya.
*   Perbedaan tahun angkatan mahasiswa mempunyai dampak dalam performa akademiknya, namun dampak tersebut tidak signifikan.







